prover/src/lib.rs

```diff
@@ -678,6 +679,12 @@ impl SP1Prover {
         }
         digest
     }
+
+    fn check_for_high_cycles(cycles: u64) {
+        if cycles > 100_000_000 {
```

Comment:
one consideration is that this could get annoying for repeated use, so we might want a env var like `DISABLE_CYCLE_COUNT_WARNING=true` to skip